<launch>
  <arg name="cars_num"/>
  <arg name="car_id"/>  
  <arg name="car_width"/>
  <arg name="car_length"/>
  
  <arg name="target_x"/>
  <arg name="target_y"/>
  <arg name="target_yaw"/>
  
  <arg name="odometry_topic"/>
  <arg name="traj_topic"/>

  <arg name="in_swarm" default="true"/>

  <arg name="max_vel"/>
  <arg name="max_acc"/>
  <arg name="max_cur"/>
  
  <!-- main node -->
  <arg name="velocity_suffix" value="1ms"/>
 <node pkg="plan_manage" type="plan_node" name="plan_node0" output="screen">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="data/mapid" value="1" type="int"/>
    <param name="data/pathid" value="1" type="int"/>
    <remap from="~traj" to="$(arg traj_topic)"/>
    <param name="odometry_topic" value="$(arg odometry_topic)"/>

    <!-- planning fsm -->
    <param name="vehicle/cars_num" value="$(arg cars_num)" />
    <param name="vehicle/car_id"  value="$(arg car_id)" />
    <param name="vehicle/car_width"  value="$(arg car_width)" />
    <param name="vehicle/car_length"  value="$(arg car_length)" />

    <param name="fsm/target_x"  value="$(arg target_x)" />
    <param name="fsm/target_y"  value="$(arg target_y)" />    
    <param name="fsm/target_yaw" value="$(arg target_yaw)" />
    <param name="fsm/in_swarm" value="$(arg in_swarm)" />

    <param name="planning/car_id"  value="$(arg car_id)" />
    <param name="planning/traj_piece_duration" value="1.0"/>
    <param name="planning/traj_res" value="8"/>
    <param name="planning/dense_traj_res" value="20"/>

  </node>

</launch>